### PR TITLE
feat/shared element

### PR DESCRIPTION
**Pull request by ethanshar**

### CODE REVIEW

File: demo/src/screens/componentScreens/SharedTransitionScreen.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+          </View>
+        </View>
+      );
+    }
+  };
+
+  render() {
+    return (
+      <View flex dark80>
+        <SharedTransition.Area renderDetails={this.renderDetails}>{this.renderList()}</SharedTransition.Area>
```
Comment (Inbal-Tish) [inline demo/src/screens/componentScreens/SharedTransitionScreen.js]: SharedTarnsition.Area must wrap the list or could it wrap just the list item?

File: demo/src/screens/componentScreens/SharedTransitionScreen.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  };
+
+  componentDidUpdate() {
+    LayoutAnimation.configureNext({
+      ...LayoutAnimation.Presets.easeInEaseOut,
+      duration: ANIMATION_DURATION,
+    });
+  }
+
+  focusItem = id => {
```
Comment (Inbal-Tish) [inline demo/src/screens/componentScreens/SharedTransitionScreen.js]: Where do you use 'focusItem()'?

File: demo/src/screens/componentScreens/SharedTransitionScreen.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    location: 'Paris, France',
+    views: 2014,
+    image:
+      'https://images.unsplash.com/photo-1544070643-24128d1f6033?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80',
+  },
+];
+
+const ANIMATION_DURATION = 200;
+
+export default class PlaygroundScreen extends Component {
```
Comment (Inbal-Tish) [inline demo/src/screens/componentScreens/SharedTransitionScreen.js]: Should be 'SharedTransitionScreen' (not 'PlaygroundScrren').

File: demo/src/screens/componentScreens/SharedTransitionScreen.js (hunk 1 above)
Comment (ethanshar) [reply demo/src/screens/componentScreens/SharedTransitionScreen.js]: it wraps the whole screen, cause it contains the shared context between the shared elements

File: demo/src/screens/componentScreens/SharedTransitionScreen.js (hunk 2 above)
Comment (ethanshar) [reply demo/src/screens/componentScreens/SharedTransitionScreen.js]: you're right, I forgot to clean the example screen. most of the code moved to the component.

File: demo/src/screens/componentScreens/SharedTransitionScreen.js (hunk 3 above)
Comment (ethanshar) [reply demo/src/screens/componentScreens/SharedTransitionScreen.js]: done

**Pull request merged**